src/libsyntax/diagnostic_list.rs

```diff
@@ -16,31 +16,374 @@
 register_long_diagnostics! {
 
 E0533: r##"
+The export_name attribute was badly formatted.
+
+Erroneous code example:
+
 ```compile_fail,E0533
-#[export_name]
+#[export_name] // error: export_name attribute has invalid format
+pub fn something() {}
+
+fn main() {}
+```
+
+The export_name attribute expects a string in order to determine the name of
+the exported symbol. Example:
+
+```
+#[export_name = "some function"] // ok!
+pub fn something() {}
+
+fn main() {}
+```
+"##,
+
+E0534: r##"
+The inline attribute was badly used.
+
+Erroneous code example:
+
+```compile_fail,E0534
+#[inline()] // error: expected one argument
+pub fn something() {}
+
+fn main() {}
+```
+
+The inline attribute can be used without arguments:
+
+```
+#[inline] // ok!
+pub fn something() {}
+
+fn main() {}
+```
+
+Or with arguments (and parens have to be used for this case!):
+
+```
+#[inline(always)] // ok!
 pub fn something() {}
 
 fn main() {}
 ```
+
+For more information about the inline attribute, take a look here:
+https://doc.rust-lang.org/reference.html#inline-attributes
+"##,
+
+E0535: r##"
+An unknown argument was given to inline attribute.
+
+Erroneous code example:
+
+```compile_fail,E0535
+#[inline(unknown)] // error: invalid argument
+pub fn something() {}
+
+fn main() {}
+```
+
+The inline attribute only knows two arguments:
+
+ * always
+ * never
+
+All other arguments given to the inline attribute will return this error.
+Example:
+
+```
+#[inline(never)] // ok!
+pub fn something() {}
+
+fn main() {}
+```
+
+For more information about the inline attribute, take a look here:
+https://doc.rust-lang.org/reference.html#inline-attributes
+"##,
+
+E0536: r##"
+No cfg-pattern was found for `not` statement.
+
+Erroneous code example:
+
+```compile_fail,E0536
+#[cfg(not())] // error: expected 1 cfg-pattern
+pub fn something() {}
+
+fn main() {}
+```
+
+The `not` statement expects at least one cfg-pattern. Example:
+
+```
+#[cfg(not(target_os = "linux"))] // ok!
+pub fn something() {}
+
+fn main() {}
+```
+
+For more information about the cfg attribute, take a look here:
+https://doc.rust-lang.org/reference.html#conditional-compilation
+"##,
+
+E0537: r##"
+A unknown predicate was used inside the cfg attribute.
+
+Erroneous code example:
+
+```compile_fail,E0537
+#[cfg(unknown())] // error: invalid predicate `unknown`
+pub fn something() {}
+
+fn main() {}
+```
+
+There are only three predicates for the cfg attribute:
+
+ * any
+ * all
+ * not
+
+Example:
+
+```
+#[cfg(not(target_os = "linux"))] // ok!
+pub fn something() {}
+
+fn main() {}
+```
+
+For more information about the cfg attribute, take a look here:
+https://doc.rust-lang.org/reference.html#conditional-compilation
+"##,
+
+E0538: r##"
+The same item was used more than once on a single element.
```

Comment:
Multiple conflicting stability annotations (the attributes) for the same item (the function/static/etc).